### ISSUE TITLE
Nested lists dont render when parent is not plain text 

### ISSUE THREAD

**Issue by tyrsius** [rexxars/react-markdown#25]
This works just fine in the [CommonMark reference tool](http://spec.commonmark.org/dingus/).

```
* Follows the [CommonMark](http://commonmark.org/) spec
* Renders actual, "native" React DOM elements
* Allows you to escape or skip HTML (try toggling the checkboxes above)
* If you escape or skip the HTML, no `dangerouslySetInnerHTML` is used! Yay
  * more things
  * stuff and things
* another list
```

This does [not render the nested list at all](http://rexxars.github.io/react-markdown/).

```
* Follows the [CommonMark](http://commonmark.org/) spec
* Renders actual, "native" React DOM elements
* Allows you to escape or skip HTML (try toggling the checkboxes above)
* If you escape or skip the HTML, no dangerouslySetInnerHTML is used! Yay
  * more things
  * stuff and things
* another list
```

This does. Any kind of markdown on the parent causes the nested list to be missed (bold, italics).

**Comment by otto-gebb** [comment]
I guess it's related. If there's a link in a list item that is not in the beginning of the line, the nested list is not rendered. See the demo below.
![markdown_bug](https://cloud.githubusercontent.com/assets/5798003/15392501/c05e81f6-1dc7-11e6-8cdd-51fe5db3dbb6.gif)

**Comment by rexxars** [comment]
Sorry for taking so long to get back to you. This was a regression caused by duplicate keys. Fixed in `2.4.1`.
